File: parent-medic-facility-service/src/main/java/org/example/entity/ParentMedicFacility.java
package org.example.entity;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;

import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Builder
@ToString
public class ParentMedicFacility {

    @Id
    private Long id;

    @Column("fk_parent_id")
    private Long fkParentId;

    @Column("fk_medic_id")
    private Long fkMedicId;

    @Column("created_at")
    private LocalDateTime createdAt;

    @Column("updated_at")
    private LocalDateTime updatedAt;

    @Column("deleted_at")
    private LocalDateTime deletedAt;
}

File: parent-medic-facility-service/src/main/kotlin/entity/ParentMedicFacility.kt
package entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import java.time.LocalDateTime

data class ParentMedicFacility(

    @Id
    @Column("id")
    var id: Long? = null,

    @Column("fk_parent_id")
    var fkParentId: Long? = null,

    @Column("fk_medic_facility_id")
    var fkMedicFacilityId: Long? = null,

    val createdAt: LocalDateTime? = null,

    val updatedAt: LocalDateTime? = null,

    val deletedAt: LocalDateTime? = null
)
